### PR TITLE
feat: delayed render

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- Adds a util hook to delay render a React component (`<Delayed/>`).
- By default a 1.5second delay. Optional custom prop with milliseconds (waitBeforeShow)
- Addresses https://github.com/cowprotocol/cowswap/issues/3005

## Todo:
- Needs more proper testing to fine tune.

### CODE REVIEW

File: src/utils/useDelayed.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React, { useState, useEffect } from 'react'
+
+type Props = {
+  children: React.ReactNode
+  waitBeforeShow?: number
+}
+
+export const Delayed = ({ children, waitBeforeShow = 1500 }: Props) => {
```
Comment (alfetopito) [inline src/utils/useDelayed.tsx:8]: Nitpick: use `PropsWithChildren`

Example: https://github.com/cowprotocol/cowswap/blob/10d798be0f843861c9c5f5380d299c23a96474cf/src/legacy/components/AnimatedDropdown/index.tsx#L8

```suggestion
import { useState, useEffect, PropsWithChildren } from 'react'

type Props = {
  waitBeforeShow?: number
}

export const Delayed = ({ children, waitBeforeShow = 1500 }: PropsWithChildren<Props>) => {
```

**Review by alfetopito — COMMENTED** [review]
Should this logic also apply to SWAP and LIMIT warnings or is the delay only needed on TWAP?

**Comment by elena-zh** [comment]
Hey @fairlighteth , looks the way better, however, I still can see green blinking banners on the form.

**Comment by fairlighteth** [comment]
> Should this logic also apply to SWAP and LIMIT warnings or is the delay only needed on TWAP?

I think it could make sense on SWAP as well, yeah.

**Comment by fairlighteth** [comment]
@alfetopito Updated and added to SWAP and LIMIT.

File: src/modules/swap/pure/warnings.tsx [hunk 2]
```diff
@@ -77,34 +78,55 @@ export const SwapWarningsTop = React.memo(function (props: SwapWarningsTopProps)
...
+      <Delayed>
+        <FeatureGuard featureFlag="advancedOrdersEnabled">
+          <TwapSuggestionBanner
+            chainId={chainId}
+            priceImpact={priceImpact}
+            buyingFiatAmount={buyingFiatAmount}
+            tradeUrlParams={tradeUrlParams}
+          />
+        </FeatureGuard>
+      </Delayed>
```
Comment (alfetopito) [inline src/modules/swap/pure/warnings.tsx:129]: This should come after the `<FeatureGuard/>` component, no?

File: src/utils/useDelayed.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+type Props = {
+  waitBeforeShow?: number
+}
+
+export const Delayed = ({ children, waitBeforeShow = 1500 }: PropsWithChildren<Props>) => {
+  const [isShown, setIsShown] = useState(false)
+
+  console.log('DELAYED RENDER: ', children)
+  console.log('waitBeforeShow: ', waitBeforeShow)
```
Comment (alfetopito) [inline src/utils/useDelayed.tsx:11]: We should remove those logs too :)

```suggestion
```

**Review by alfetopito — COMMENTED** [review]
I feel like there's too much repetition and we'll for sure forget to add this when adding a new warning to a component. Could this be instead added to the a base warning component?
Not all have a same root, but at least some do and this could reduce the repetition.

**Comment by fairlighteth** [comment]
> I feel like there's too much repetition and we'll for sure forget to add this when adding a new warning to a component. Could this be instead added to the a base warning component? Not all have a same root, but at least some do and this could reduce the repetition.

@alfetopito I think that's a good point. I know re-factored the code to use `<Delayed>` as much as possible in the root banner/warning components. This indeed simplifies the import and usage of banner/warning components.

**Pull request closed without merging**